### PR TITLE
[C-6761] adds preference Sections view

### PR DESCRIPTION
**Pull request by heyglassy**
## Description

Added Preference Section view to React-Preferences

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> `Fix [#1]()`

### CODE REVIEW

File: packages/client-graphql/src/preferences.ts [hunk 1]
```diff
@@ -15,6 +15,24 @@ const RECIPIENT_PREFERENCES = `
   }
 `;
 
+const PREFERENCE_SECTIONS = `
+query GetPreferenceSections{
+	preferenceSections {
+		nodes {
```
Comment (suhasdeshpande) [inline packages/client-graphql/src/preferences.ts:21]: indentation seems off??

File: packages/react-preferences/src/components/ChannelPreferences.tsx [hunk 2]
```diff
@@ -67,10 +71,10 @@ export const ChannelPreferences: React.FC<{
     onPreferenceChange({
       hasCustomRouting: !checked,
```
Comment (suhasdeshpande) [inline packages/react-preferences/src/components/ChannelPreferences.tsx:72]: we moved this flag to section level now

File: packages/react-preferences/src/components/PreferenceList.tsx [hunk 3]
```diff
@@ -13,15 +13,91 @@ export const StyledList = styled.div`
...
+          <PreferenceSection
+            section={section}
+            preferences={preferences}
+            key={key}
+          />
+        );
+      });
+    }
+
+    if (preferences.preferenceSections?.length < 1) {
```
Comment (suhasdeshpande) [inline packages/react-preferences/src/components/PreferenceList.tsx:84]: this is to maintain backward compatibility?

File: packages/react-preferences/src/components/PreferenceList.tsx [hunk 4]
```diff
@@ -13,15 +13,92 @@ export const StyledList = styled.div`
...
+
 export const PreferenceList: React.FunctionComponent<{
   // TODO: define Preferences theming
   theme?: ThemeProps<any>;
 }> = (props) => {
   const { brand } = useCourier();
   const preferences = usePreferences();
 
+  const renderPreferences = () => {
+    if (preferences?.isLoading) {
```
Comment (suhasdeshpande) [inline packages/react-preferences/src/components/PreferenceList.tsx:68]: do we want to be checking if sections are loading?

Comment (heyglassy) [reply packages/react-preferences/src/components/PreferenceList.tsx:68]: This checks if everything getting loaded into preferences, including sections, is being loaded. So it covers that base.

**Review by suhasdeshpande — APPROVED** [review]

**Pull request merged**